### PR TITLE
test: add test cases to increase coverage for util

### LINKED ISSUE #719: improve ut coverage for util.go
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

The line in red is missing ut test coverage:
https://coveralls.io/builds/36878386/source?filename=pkg%2Futil%2Futil.go

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by sunpa93**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #719 

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [x] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
```
run golangci-lint
  Running [/home/runner/golangci-lint-1.29.0-linux-amd64/golangci-lint run --out-format=github-actions -E=gofmt --timeout=30m0s -E=deadcode -E=unused -E=varcheck -E=ineffassign] in [] ...
  Error: Error return value of `os.Mkdir` is not checked (errcheck)
  
  Error: issues found
```

File: pkg/util/util_test.go [hunk 1]
```diff
@@ -125,23 +126,143 @@ func TestConvertTagsToMap(t *testing.T) {
...
+				err := os.Mkdir("./target_test", os.FileMode(0755))
+				assert.NoError(t, err, "Failed in setup: %v", err)
+			},
+			targetDir:     "./target_test",
+			expectedError: false,
+		},
+		{
+			desc: "[Error] existing file in target path",
+			setup: func() {
+				_, err := os.Create("./file_exists")
```
Comment (edreed) [inline pkg/util/util_test.go:153]: You'll need to call `Close()` on the returned `File` object in order to be able to delete it on line 159 below. I believe this is the reason the Windows unit tests are failing to delete the file with an "in-use" error.

**Review by edreed — CHANGES_REQUESTED** [review]

File: pkg/util/util_test.go (hunk 1 above)
Comment (sunpa93) [reply pkg/util/util_test.go]: Ah! Thank you for pointing this out. I forgot about closing the file pointer 👍

File: pkg/util/util_test.go [hunk 2]
```diff
@@ -125,23 +126,144 @@ func TestConvertTagsToMap(t *testing.T) {
...
+		},
+		{
+			desc: "should not return error if path already exists",
+			setup: func() {
+				err := os.Mkdir("./target_test", os.FileMode(0755))
+				assert.NoError(t, err, "Failed in setup: %v", err)
+			},
+			targetDir:     "./target_test",
+			expectedError: false,
+		},
```
Comment (edreed) [inline pkg/util/util_test.go:149]: You should remove the directory after the test case has run just like in `TestMakeFile`.

**Review by edreed — CHANGES_REQUESTED** [review]
Noticed one more thing.

File: pkg/util/util_test.go (hunk 2 above)
Comment (sunpa93) [reply pkg/util/util_test.go:149]: The removal should be handled by the following block in this case:

https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/743802637524d7759b70c08979211a12b7c0bb37/pkg/util/util_test.go#L173-L177

Comment (edreed) [reply pkg/util/util_test.go:149]: Oh missed that. Thanks for pointing it out!

**Review by edreed — APPROVED** [review]

**Review by andyzhangx — APPROVED** [review]
/approve

**Pull request merged**